sql/hive/src/test/scala/org/apache/spark/sql/hive/InsertSuite.scala

```diff
@@ -769,6 +769,30 @@ class InsertSuite extends QueryTest with TestHiveSingleton with BeforeAndAfter
     }
   }
 
+  test("SPARK-31459: fix insert overwrite directory target path is an existing file") {
+    withTempView("test_insert_table") {
+      spark.range(10).selectExpr("id", "id AS str").createOrReplaceTempView("test_insert_table")
+
+      withTempDir { dir =>
+        val tempFile = File.createTempFile("targetPath", "", dir)
+        val path = tempFile.toURI.getPath
+
+        sql(
+          s"""
+            |INSERT OVERWRITE LOCAL DIRECTORY '${path}'
```

Comment:
Please remove this misleading test case. Apache Spark should not delete the existing file.

Reply:
There is no problem at `INSERT OVERWRITE LOCAL DIRECTORY` syntax.

Reply:
ok